spline/components/tasks.py

```diff
@@ -218,6 +219,32 @@ def can_process_shell(self, entry):
 
     def process_shell(self, creator, entry, config):
         """Processing a shell entry."""
+        retries = entry['retries'] if 'retries' in entry else 1
+        counter = 0
+        output = []
+
+        while counter < retries:
+            result = Adapter(self._process_shell(creator, entry, config))
+            output += result.output
+
+            if not result.success:
+                counter = counter + 1
+
+                if counter < retries:
+                    to_sleep = counter * counter
```

Comment:
Why have you choosen to wait by that  ... formular?
1, 4, 9, 16 .... talking about seconds

Reply:
Exponential back-off is sort of the typical way this is handled.